packages/itwin/map-layers/src/ui/widget/MapLayerManager.tsx

```diff
@@ -248,17 +234,25 @@ export function MapLayerManager(props: MapLayerManagerProps) {
         return;
       }
 
-      // This is where the list of layers first gets populated... I need to update it
-      // MapUrlDialog gets around knowing MapLayerManager exists and vice versa by affecting the viewports displayStyle which MapLayerManager is listening for
-      // We know when displayStyle changes we've added a layer, this layer may not be a custom layer
-      sourceLayers?.layers.forEach((source: MapLayerSource) => { sources.push(source); });
+      // This is where the list of layers first gets populated...
+      const sources: MapLayerSource[] = [];
+      const bases: MapLayerSource[] = [];
+      const addSource = (source: MapLayerSource) => source.baseMap ? bases.push(source) : sources.push(source);
+      sourceLayers?.allSource.forEach(addSource);
+      preferenceSources.forEach((source) => {
+        // Do not add duplicate
+        if (!sources.find((curSource) => source.name === curSource.name))
+          addSource(source);
+      });
+      sources.sort((a: MapLayerSource, b: MapLayerSource) => compareStrings(a.name.toLowerCase(), b.name.toLowerCase()));
```

Comment:
What is the purpose of the sort here? Just trying to understand

Reply:
Simply to keep sources sorted in the UI.

Reply:
@eringram please resolve